scripts/common_startup_functions.sh

```diff
@@ -187,9 +187,14 @@ find_server() {
         fi
         server_args="$server_args $uwsgi_args"
     elif [ "$APP_WEBSERVER" = "gunicorn" ]; then
-        export GUNICORN_CMD_ARGS="${GUNICORN_CMD_ARGS:-\"--bind=localhost:8080\"}"
         run_server="gunicorn"
-        server_args="'galaxy.webapps.${server_app}.fast_factory:factory()' --pythonpath lib -k ${gunicorn_worker:-$default_gunicorn_worker} $gunicorn_args"
+        if [ "$server_app" = "tool_shed" ]; then
+            server_args="'${server_app}.webapp.fast_factory:factory()' --pythonpath lib -k ${gunicorn_worker:-$default_gunicorn_worker} $gunicorn_args"
+            export GUNICORN_CMD_ARGS="${GUNICORN_CMD_ARGS:-\"--bind=localhost:9009\"}"
+        else
+            export GUNICORN_CMD_ARGS="${GUNICORN_CMD_ARGS:-\"--bind=localhost:8080\"}"
+            server_args="'galaxy.webapps.${server_app}.fast_factory:factory()' --pythonpath lib -k ${gunicorn_worker:-$default_gunicorn_worker} $gunicorn_args"
+        fi
```

Comment:
I think we can simplify this by setting the port in one place. We set the port for reports on line 166; if port selection is not dependent on which server will run, let's just move port selection out of this if-statement. `server_args`, on the other hand, is server-dependent, so it should stay here; and aren't we missing setting `server_args` for reports in the same way we are setting it for tool_shed?

Reply:
I've moved the port setting. Reports is in galaxy.webapps, so that is correct.